src/streams.h

```diff
@@ -565,7 +565,7 @@ class CBufferedFile
 
     // check whether we're at the end of the source file
     bool eof() const {
-        return nReadPos == nSrcPos && feof(src);
+        return nReadPos == nSrcPos && src != NULL && feof(src);
```

Comment:
The only way that `src` can be set to `NULL` while we still have a `CBufferedFile` around is if `CBufferedFile.fclose()` has been called. This will result in `CBufferedFile.eof()` returning false. Is this the behaviour we want here?